src/Form/DataMembershipType.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Form;
+
+use App\DataForm\DataMembership;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\File;
+
+class DataMembershipType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('membershipFile', FileType::class, [
+                'label' => 'Bulletin d\'adhésion (PDF) : ',
+                'required' => true,
+                'constraints' => [
+                    new File([
+                        'maxSize' => '512k',
+                        'mimeTypes' => [
+                            'application/pdf'
+                        ],
+                        'mimeTypesMessage' => 'Veuillez enregistrer un PDF valide.',
+                    ])
+                ]
```

Comment:
You have a Form data class (membership), it'll be easier to put your constaints there using anotations.

Reply:
My beginner's blunders...